samples/sample-04/src/main/kotlin/com/remusrd/notesample/configuration/KafkaConfig.kt

```diff
@@ -0,0 +1,57 @@
+package com.remusrd.notesample.configuration
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder
+import org.springframework.kafka.core.*
+import org.springframework.kafka.support.serializer.JsonDeserializer
+import org.springframework.kafka.support.serializer.JsonSerializer
+
+@Configuration
+class KafkaConfig {
+    @Autowired
+    lateinit var jackson2ObjectMapperBuilder: Jackson2ObjectMapperBuilder
+    @Autowired
+    lateinit var kafkaProperties: KafkaProperties
+
+
+    @Bean
+    fun kafkaTemplate(): KafkaTemplate<Any, Any> {
+        return KafkaTemplate<Any, Any>(defaultKafkaProducerFactory())
+    }
+
+
+    @Bean
+    fun defaultKafkaConsumerFactory(): ConsumerFactory<Any, Any> {
```

Comment:
Why do we need to override auto-configured beans?
This one and the next - `defaultKafkaProducerFactory()`.

Reply:
What should I do instead?, creating my own beans without replacing the default ones?